src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -173,23 +173,30 @@ public static Version PSVersion
         {
             get
             {
-                return (SemanticVersion)GetPSVersionTable()[PSVersionInfo.PSVersionName];
+                if (s_psVersionCache == null)
+                {
+                    // Ideally, we only do one conversion for 'PSVersion' and use the cached value afterwards.
+                    var version = (Version)(SemanticVersion)s_psVersionTable[PSVersionInfo.PSVersionName];
+                    Interlocked.CompareExchange(ref s_psVersionCache, version, null);
```

Comment:
It seems we always use the value. So we could initialize it in `static PSVersionInfo()`.

Reply:
That makes sense. Moved to the type initializer.